Functions/Admin/Backup-RsEncryptionKey.ps1

```diff
@@ -4,71 +4,134 @@
+	        
+	    .PARAMETER KeyPath
+	        Specify the path to where the encryption key should be stored. 
+
+	    .EXAMPLE
+	        Backup-RSEncryptionKey -Password 'Enter Your Password' -KeyPath 'C:\ReportingServices\Default.snk'
+	        Description
+	        -----------
+	        This command will back up the encryption key against default instance from SQL Server 2016 Reporting Services
+	    
+	    .EXAMPLE
+	        Backup-RSEncryptionKey -ReportServerInstance 'SQL2012' -ReportServerVersion '11' -Password 'Enter Your Password' -KeyPath 'C:\ReportingServices\Default.snk'
+	        Description
+	        -----------
+	        This command will back up the encryption key against named instance (SQL2012) from SQL Server 2012 Reporting Services
+	
+		.NOTES
+			Author:      ???
+			Editors:     Friedrich Weinmann
+			Created on:  ???
+			Last Change: 31.01.2017
+			Version:     1.1
+	
+			Release 1.1 (26.01.2017, Friedrich Weinmann)
+			- Fixed Parameter help (Don't poison the name with "(optional)", breaks Get-Help)
+			- Implemented ShouldProcess (-WhatIf, -Confirm)
+			- Replaced calling exit with throwing a terminating error (exit is a bit of an overkill when failing a simple execution)
+			- Improved error message on failure.
+			- Standardized the parameters governing the Report Server connection for consistent user experience.
+	
+			Release 1.0 (???, ???)
+			- Initial Release
     #>
 
-    [cmdletbinding()]
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
     param
     (
+        [Alias('SqlServerInstance')]
         [string]
-        $SqlServerInstance='MSSQLSERVER',
-
+        $ReportServerInstance,
+        
+        [Alias('SqlServerVersion')]
+        [ReportingServicesTools.SqlServerVersion]
+        $ReportServerVersion,
+        
         [string]
-        $SqlServerVersion='13',
+        $ComputerName,
+        
+        [System.Management.Automation.PSCredential]
+        $Credential,
 
-        [Parameter(Mandatory=$True)]
+        [Parameter(Mandatory = $True)]
         [string]
         $Password,
         
-        [Parameter(Mandatory=$True)]
+        [Parameter(Mandatory = $True)]
         [string]
-        $KeyPath    
-    )
-
-    $rsWmiObject = New-RsConfigurationSettingObject -SqlServerInstance $SqlServerInstance -SqlServerVersion $SqlServerVersion
-
-    Write-Verbose "Retrieving encryption key..."
-    $encryptionKeyResult = $rsWmiObject.BackupEncryptionKey($Password)
-
-    if ($encryptionKeyResult.HRESULT -eq 0) 
-    {
-        Write-Verbose "Success!"
-    }
-    else 
-    {
-        Write-Error "Fail! `n Errors: $($encryptionKeyResult.ExtendedErrors)"
-        Exit 1
-    }
-
-    Write-Verbose "Writing key to file..."
-    [System.IO.File]::WriteAllBytes($KeyPath, $encryptionKeyResult.KeyFile)
-    Write-Verbose "Success!"
+        $KeyPath
+	)
+    
+    if ($ComputerName) { $tempComputerName = $ComputerName }
```

Comment:
While I'd argue against it improving readability on single-line conditional actions predicated by a simple existence test such as you highlighted - rather I think it bloats line numbers within the actual code and makes it harder to read - if the official decision is to do that for _all_ if/else conditional actions, no matter what, I'll have that updated. You sure?